HW04.Rmd

```diff
@@ -18,44 +18,97 @@ output: github_document
 The gun_deaths.csv is provided in the repository. 
 ```{r load-data, message=FALSE, warning=FALSE}
 library(tidyverse)    # load tidyverse packages, including ggplot2
+library(devtools)
+library(knitr)
+library(kableExtra)
 
 # read in the data
+gun_deaths_data <- read.csv("gun_deaths.csv")
+
+#glimpsing at the data, couldn't use function "glimpse" for some reason
+head(gun_deaths_data)
+
+tail(gun_deaths_data)
+
+summary(gun_deaths_data)
+
+
+#changing label of months to three letter codes from digits 1-12
+gun_deaths_data$month <- factor(gun_deaths_data$month, labels = c("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec" ))
+
+#changing race label
+gun_deaths_data$race <- gsub(pattern = "Not Availabletive American/Not Availabletive Alaskan", 
+     replacement = "Native American/Native Alaskan",
+     x = gun_deaths_data$race)
 
 ```
 
+
+
 ## Generating a data frame that summarizes the number of gun deaths per month and printing using `kable()`. 
 
 Knitr is part of the knitr package. Allows you to change column names and add captions and make pretty tables in your knitted document. Hint: set the argument format = "markdown"
 
 
 ```{r month, warning=FALSE}
+gun_deaths_per_month <- gun_deaths_data %>%
+    group_by(month) %>%
+    count()
 
+kable(gun_deaths_per_month, format = "markdown", col.names = c("Month", "# of Gun Deaths"), caption = "number of deaths by gun in each month from 2012-2014, sourced from FiveThirtyEight")
 ```
 
 
 ### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
 
 ```{r month_chart}
-
+ggplot(gun_deaths_data, aes(month)) +
+           scale_y_continuous(breaks = seq(0, 10000, by = 1000)) +
+           geom_bar(fill = "orchid4") +
+               labs(x = "Month", y = "Number of Gun Deaths", title = "Total Gun Deaths per Month from 2012-2014", caption = "Sourced from FiveThirtyEight") +
+               theme_light()
+           
 ```
 
 ## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
 
 ```{r intent}
+ggplot(gun_deaths_data, aes(fct_infreq(intent))) +

```

Comment:
Oh wow. I didn't know fct_infreq existed. What a neat function (is that what they're called? lol). So much better than inputting a specific order.